src/main/java/com/exasol/projectkeeper/validators/changelog/dependencies/MavenDependencyComparator.java

```diff
@@ -0,0 +1,66 @@
+package com.exasol.projectkeeper.validators.changelog.dependencies;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import org.apache.maven.model.Dependency;
+
+/**
+ * This class compares two lists of maven dependencies.
+ */
+public class MavenDependencyComparator {
+
+    /**
+     * Compare two lists of maven dependencies.
+     * 
+     * @param oldDependencies old list of dependencies
+     * @param newDependencies new list of dependencies
+     * @return list of dependency changes
+     */
+    public List<DependencyChange> compare(final List<Dependency> oldDependencies,
```

Comment:
Can you please extract small methods here? It's hard to understand what's going on